x-pack/plugins/apm/server/routes/fleet/get_latest_apm_package.ts

```diff
@@ -17,12 +17,19 @@ export async function getLatestApmPackage({
   request: KibanaRequest;
 }) {
   const packageClient = fleetPluginStart.packageService.asScoped(request);
-  const { name, version } = await packageClient.fetchFindLatestPackage(
+  const latestPackage = await packageClient.fetchFindLatestPackage(
     APM_PACKAGE_NAME
   );
-  const registryPackage = await packageClient.getPackage(name, version);
-  const { title, policy_templates: policyTemplates } =
-    registryPackage.packageInfo;
+  const packageInfo =
+    'buffer' in latestPackage
+      ? (await packageClient.readBundledPackage(latestPackage)).packageInfo
+      : latestPackage;
+  const {
+    name,
+    version,
+    title,
+    policy_templates: policyTemplates,
+  } = packageInfo;
```

Comment:
Is the fleet migration covered by an API test? If not, can you add one?

Reply:
will do

Reply:
I added a suite of tests for the migration check affected by the fix in this PR. The tests required a new FTR config `cloud` and a limited need for a new testing user `apm_manage_service_account` to test a specific edge case in permissions. There's still opportunity to expand these tests further beyond the migration check logic, perhaps while executing the next test plan.